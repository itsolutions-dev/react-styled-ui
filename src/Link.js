import styled from 'styled-components';
import {
  getTextColor,
  getBackgroundColor,
  getOpacity,
  getHover,
} from './utils/';

const Link = styled.a`
  margin: 0 1em;
  background-color: ${props => props.transparent === true ? 'transparent' : getBackgroundColor(props)};
  color: ${props => getTextColor(props)};
  opacity: ${props => getOpacity(props)};

  &:hover {
    background-color: ${props => props.transparent === true ? 'transparent' : getHover(props)};
  }
`;

export default Link;
