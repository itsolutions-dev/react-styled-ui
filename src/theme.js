export default {
  radius: 8,
  spacing: {
    margin: 8,
    padding: 8,
    iconSize: 24,
    borderSize: 1,
    shadowSize: 1,
  },
  font: {
    size: 16,
    weight: 200,
    lineHeight: 18,
    family: 'Roboto, sans-serif',
  },
  h1: {
    size: 32,
    weight: 900,
    lineHeight: 64,
  },
  h2: {
    size: 24,
    weight: 700,
    lineHeight: 32,
  },
  h3: {
    size: 18,
    weight: 500,
    lineHeight: 24,
  },
  selected: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  hover: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  transitions: {
    in: undefined,
    out: undefined,
  },
  palette: {
    default: 'green',
    defaultText: 'white',
    defaultDisabled: undefined,
    defaultHover: undefined,
    defaultSelected: undefined,
    primary: 'red',
    primaryText: 'white',
    primaryDisabled: undefined,
    primaryHover: undefined,
    primarySelected: undefined,
    secondary: 'blue',
    secondaryText: 'white',
    secondaryDisabled: undefined,
    secondaryHover: undefined,
    secondarySelected: undefined,
    disabled: 'lightgray',
    shadow: 'black',
  },
};
