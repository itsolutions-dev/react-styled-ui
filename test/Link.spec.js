import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import theme from '../src/theme';
import Link from '../src/Link';

describe('Link', () => {
  test('should render a Link', () => {
    const rendered = ReactTestUtils.renderIntoDocument(<Link theme={theme} />);
    const component = ReactTestUtils.findRenderedDOMComponentWithTag(
      rendered,
      'a',
    );
    expect(component).toBeDefined();
  });
});
