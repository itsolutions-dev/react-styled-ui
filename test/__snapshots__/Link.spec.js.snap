// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render a Link 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 16px;
  color: #4c4c4c;
  opacity: 1;
}

.c0:hover {
  color: #f5f7f7;
}

<a
  className="c0"
/>
`;
